src/data/firestore/partenership/models.ts

```diff
@@ -0,0 +1,37 @@
+import type { FirestoreDataConverter, QueryDocumentSnapshot } from "firebase-admin/firestore";
+import type { FirestoreDocument } from "~/data/model-document-mapper";
+import type { Partner, Partnership } from "~/models/partnership/partnership.type";
+
+type PartnershipDoc = FirestoreDocument<Partnership>;
+type PartnerDoc = FirestoreDocument<Partner>;
+
+const partnerDataConverter: FirestoreDataConverter<Partner, PartnerDoc> = {
+    fromFirestore: (snapshot: QueryDocumentSnapshot) => {
+        const data = snapshot.data() as PartnerDoc;
+        data.id = snapshot.id;
+
+        return { ...data };
+    },
+    toFirestore: (model: Partner) => {
+        delete model.id;
+        return { ...model };
+    }
```

Comment:
potresti cambiare anche questo mettendo prima il toFirestrore. It is not necessary but only to standardize converters

Reply:
I fixed the requested changes and I wrote the test.  Have I put on `/test` directory? For now I put it in `/test/partnership.test.ts`